tests/js/client/communication/test-communication.js

```diff
@@ -215,24 +263,537 @@ function CommunicationSuite () {
+      case "Graph":
+        return `db._query("FOR v IN ${vertexName} FOR t IN 1 OUTBOUND v ${edgeName} FOR x IN 1..${docsPerWrite} INSERT {} INTO ${twoShardColName} OPTIONS {exclusive: false}")`;
+      case "NamedGraph":
+        return `db._query("FOR v IN ${vertexName} FOR t IN 1 OUTBOUND v GRAPH ${graphName} FOR x IN 1..${docsPerWrite} INSERT {} INTO ${twoShardColName} OPTIONS {exclusive: false}")`;
+      case "View":
+        return `db._query("FOR v IN ${viewName} OPTIONS {waitForSync: true} FOR x IN 1..${docsPerWrite} INSERT {} INTO ${twoShardColName} OPTIONS {exclusive: false}")`;
+      case "Satellite":
+        return `db._query("FOR v IN ${vertexName} FOR x IN 1..${docsPerWrite} INSERT {} INTO ${twoShardColName} OPTIONS {exclusive: true}")`;
+        default:
+        // Illegal Test
+        assertEqual(true, false);
+    }
+  };
+
+  const selectReadQuery = () => {
+    switch (type) {
+      case "Plain":
+        return `db._query("FOR x IN ${twoShardColName} RETURN x")`;
+      case "Graph":
+        return `db._query("FOR v IN ${vertexName} FOR t IN 1 OUTBOUND v ${edgeName} FOR x IN ${twoShardColName} RETURN x")`;
+      case "NamedGraph":
+        return `db._query("FOR v IN ${vertexName} FOR t IN 1 OUTBOUND v GRAPH ${graphName} FOR x IN ${twoShardColName} RETURN x")`;
+      case "View":
+        return `db._query("FOR v IN ${viewName} OPTIONS {waitForSync: true} FOR x IN ${twoShardColName} RETURN x")`;
+      case "Satellite":
+        return `db._query("FOR v IN ${vertexName} FOR x IN ${twoShardColName} RETURN x")`;
+      default:
+        // Illegal Test
+        assertEqual(true, false);
+    }
+  };
+
+  const exclusiveQuery = selectExclusiveQuery();
+  const writeQuery = selectWriteQuery();
+  const readQuery = selectReadQuery();
+  const jsWriteAction = `
+    function() {
+      const db = require("@arangodb").db;
+      const col = db.${twoShardColName};
+      for (let i = 0; i < ${docsPerWrite}; ++i) {
+        col.save({});
+      }
+    }
+  `;
+  const jsReadAction = `
+  function() {
+    const db = require("@arangodb").db;
+    const col = db.${twoShardColName};
+    let result = col.toArray();
+    return result;
+  }
+`;
+  const jsExclusive = `
+    db._executeTransaction({
+      collections: {
+        exclusive: "${twoShardColName}"
+      },
+      action: ${jsWriteAction}
+    });
+  `;
+  const jsWrite = `
+    db._executeTransaction({
+      collections: {
+        write: "${twoShardColName}"
+      },
+      action: ${jsWriteAction}
+    });
+  `;
+  const jsRead = `
+    db._executeTransaction({
+      collections: {
+       read: "${twoShardColName}"
+      },
+      action: ${jsReadAction}
+    });
+  `;
+
+  const apiLibs = `
+    const request = require("@arangodb/request");
+    arango.getEndpoint()
+    function sendRequest(method, endpoint, body, headers) {
+
+      const endpointToURL = (endpoint) => {
```

Comment:
Include the method from above here as a variable?

Reply:
i am not sure if this is possible or more readable.
With this method we need to send a working string representation of a JS program to the server.
If we pull in the variable here, the stringifaction process may do some odd things to it.
I will try it.